packages/mongodb-rag-ingest/src/embed/updateEmbeddedContent.test.ts

```diff
@@ -138,4 +146,64 @@ describe("updateEmbeddedContent", () => {
       "2cbfe9901657ca15260fe7f58c3132ac1ebd0d610896082ca1aaad0335f2e3f1"
     );
   });
-});
+  it("processes pages concurrently", async () => {
+    const pageStore = makeMockPageStore();
+    const concurrentPages: Page[] = [
+      { ...examplePage, url: 'https://example.com/test1'},
+      { ...examplePage, url: 'https://example.com/test2'},
+      { ...examplePage, url: 'https://example.com/test3'}
+    ];
+
+    await persistPages({
+      pages: concurrentPages,
+      store: pageStore,
+      sourceName: "test",
+    });
+
+    const embeddedContentStore = makeMockEmbeddedContentStore();
+
+    const since = new Date("2000-01-01");
+
+    const startTimes: number[] = [];
+    const endTimes: number[] = [];
+
+    const originalDateNow = Date.now;
+    let currentFakeTime = originalDateNow();
+
+    jest.spyOn(global.Date, 'now').mockImplementation(() => currentFakeTime);
+
+    const mockEmbedder: jest.Mocked<Embedder> = {
+      embed: jest.fn().mockImplementation(async (param) => {
+        const startTime = Date.now();
+        startTimes.push(startTime);
+        await new Promise((resolve) => setTimeout(resolve, Math.random() * 100)); // Random delay to simulate processing
```

Comment:
why not just do a static amount, such as 50ms?

```suggestion
        await new Promise((resolve) => setTimeout(resolve, 50));
```